internal/types/infrastructure/validation.go

```diff
@@ -99,6 +99,25 @@ func (i *InstanceRequest) Validate() error {
 	if i.Image == "" {
 		return fmt.Errorf("image is required")
 	}
+
+	// Validate payload if provided
+	if i.Payload != "" && len(i.Payload) > 1024*1024*100 { // 100MB limit
+		return fmt.Errorf("payload exceeds maximum size of 100MB")
+	}
+
+	// Validate payload path if provided
+	if i.PayloadPath != "" {
+		// Ensure path starts with / (absolute path)
+		if !strings.HasPrefix(i.PayloadPath, "/") {
+			return fmt.Errorf("payload_path must be an absolute path (start with /)")
+		}
+
+		// Basic path validation - no relative path components
+		if strings.Contains(i.PayloadPath, "..") {
+			return fmt.Errorf("payload_path cannot contain relative path components (..)")
+		}
```

Comment:
To help prevent user errors we can also call [Clean](https://pkg.go.dev/path/filepath#Clean) at the API level.